### PR TITLE
Fix measure with condition display in mpl and latex drawers

### LINKED ISSUE #6804: Problem in drawing circuit for conditional measurement using multiple classical registers
### Information
The `qc.draw()` function does not draw the quantum circuit properly if we change the order of classical registers while initializing the `QuantumCircuit()` instance if there is a measurement in the circuit that is conditioned on the output of some other classical register.

- **Qiskit Terra version**:
```
{'qiskit-terra': '0.17.4', 'qiskit-aer': '0.8.2', 'qiskit-ignis': '0.6.0', 'qiskit-ibmq-provider': '0.14.0', 'qiskit-aqua': '0.9.2', 'qiskit': '0.27.0', 'qiskit-nature': '0.1.3', 'qiskit-finance': None, 'qiskit-optimization': None, 'qiskit-machine-learning': None}
```
- **Python version**: 3.8.6
- **Operating system**: MacOS

### What is the current behavior (and how to reproduce the problem)?
Screenshot 2021-07-26 at 1 48 35 PM

Notice that in the above code, we initialize the quantum circuit as:
```py
qc = QuantumCircuit(q_a, q_b, c_a, c_b)
```

However, if we exchange the classical registers while initializing, like so:
```py
qc = QuantumCircuit(q_a, q_b, c_b, c_a)
```
we get this, which is the correct circuit...
Screenshot 2021-07-26 at 1 48 55 PM

### What is the expected behavior?
The circuit in the first image should have the arrows going to the classical register `c_b` and the dots going to `c_a`, but here both of them go to `c_a`.

### PR DESCRIPTION
**Pull request by enavarro51**
### Summary

Fixes #6804

### Details and comments

This PR fixes a problem where a measure with condition will sometimes display incorrectly in the 'mpl' and 'latex' drawers. The corrected displays are below.

```
qr = QuantumRegister(2, "qr")
cr1 = ClassicalRegister(2, "cr1")
cr2 = ClassicalRegister(2, "cr2")
qc = QuantumCircuit(qr, cr1, cr2)
qc.h(0)
qc.h(1)
qc.measure(0, cr1[0])
qc.measure(1, cr2[0]).c_if(cr1, 1)
qc.draw('mpl')
```
image
image

### CODE REVIEW

**Comment by TharrmashasthaPV** [comment]
Hi @enavarro51 . Nice one! I think supporting measure with condition in latex drawer has some issues. For instance, when `cregbundle=False`, then the same circuit given above returns as below:

latex_6822

You can notice here that it is hard to distinguish between the conditional bits and the bit on which the measurement is stored. I guess we would be able to overcome this issue if we had different notations for the conditional bits and the storage bit. I understand that it is a completely different issue that what this PR is for. But any opinions on that?

**Comment by enavarro51** [comment]
Good point. It would be nice if there were some kind of arrow so it mimics the mpl drawer. Not sure how that works in Latex.
image

**Comment by enavarro51** [comment]
@TharrmashasthaPV How does this look? Not perfect, but I think it solves your issue.

image

**Comment by TharrmashasthaPV** [comment]
> @TharrmashasthaPV How does this look? Not perfect, but I think it solves your issue.
> 
> image

@enavarro51 This certainly is better than having just dots. But I was thinking we could do something like 

latex_6822

I guess replacing `\\cwx[#1]` with `\\ar @{<=} [#1,0]` should do the trick if I am not wrong. Also in #6447 we were trying to get a minimal latex source code with minimal headers. So, I guess this way we can also be in line with its goal.😃 What do you think?

**Comment by enavarro51** [comment]
@TharrmashasthaPV Very nice! Thought you might have a trick to use. I'll drop it in. Seems to work in all the cases I've tried. I added it to the cregbundle True as well. Is there a way to add a bullet in addition to the '= 1' at the condition creg?

image

**Comment by TharrmashasthaPV** [comment]
> @TharrmashasthaPV Very nice! Thought you might have a trick to use. I'll drop it in. Seems to work in all the cases I've tried. I added it to the cregbundle True as well. Is there a way to add a bullet in addition to the '= 1' at the condition creg?
> 
> image

Awesome. For having both the bullet and `=1`, I think currently we use ` \dstick{_{_{\hspace{1.5em}=1}}} \cw \cwx[-1]` which just gives `=1`. But alternatively we can use something like `\control \cw^(-0.25){^{=1}} \cwx[-1]` to have both. The parameter that has been set to `-0.25` can be adjusted as per the need I guess.

**Comment by enavarro51** [comment]
Thanks a lot, @TharrmashasthaPV. Sometimes it's great to have someone else write your code for you. 😊 Anyway, I've extended this so the mpl and latex drawers should now show the same information for measures and conditionals. I'm not sure why mpl uses hex for the condition and the other 2 drawers don't. Maybe a discussion for another time. You may need to incorporate these changes in #6248 and #6259 at some point.
image
image
image
image

**Comment by enavarro51** [comment]
The latex and text drawers now show conditional values in hex, the same as the mpl drawer.

image
```
       ┌───┐┌─┐   ┌───┐
 qr_0: ┤ H ├┤M├───┤ H ├
       ├───┤└╥┘┌─┐└─╥─┘
 qr_1: ┤ H ├─╫─┤M├──╫──
       └─╥─┘ ║ └╥┘  ║  
cr1_0: ══o═══╬══╬═══╬══
         ║   ║  ║   ║  
cr1_1: ══■═══╩══╬═══╬══
        0x2     ║   ║  
                ║   ║  
cr2_0: ═════════╩═══■══
                    ║  
cr2_1: ═════════════■══
                   0x3 
```

**Review by javabster — COMMENTED** [review]
I'm a bit unsure why there is a new image for `user_ax.png`?

**Comment by enavarro51** [comment]
Not a great reason. After a discussion with Luciano, I added the line onto the previous empty plot to make it more realistic. Not too important either way.

**Comment by javabster** [comment]
> Not a great reason. After a discussion with Luciano, I added the line onto the previous empty plot to make it more realistic. Not too important either way.

Ok, but how is it relevant to the work in this PR?

**Comment by enavarro51** [comment]
Uh, not really. Would you like it removed?

**Comment by javabster** [comment]
Also, could you add a release note please 😄

**Comment by javabster** [comment]
> Uh, not really. Would you like it removed?

Ah ok. Generally it's preferable to keep the content of PRs as relevant as possible to the original issue, and open new issues to cover extra things, but in this case it's quite small and not worth opening a whole new PR for it  (and there is already scope creep with the switching to hex conditionals as well anyway) so I think just leave it

**Comment by enavarro51** [comment]
Oh, 'scope creep'. I like that. 😊 Will do the release note.

**Comment by enavarro51** [comment]
Release note added in https://github.com/Qiskit/qiskit-terra/pull/6822/commits/056eb8b3e8bbb323c42318138c2646edb92a9bc9.

**Review by javabster — APPROVED** [review]
This lgtm! Thanks for working on this, just a few merge conflicts left and I think it'll be good to go

**Comment by enavarro51** [comment]
@javabster Merge conflicts fixed in https://github.com/Qiskit/qiskit-terra/pull/6822/commits/3bd0c3ed27b66762b0fc7a15145562a458fddefc.

**Review by javabster — APPROVED** [review]
LGTM 🚀

File: qiskit/visualization/latex.py [hunk 1]
```diff
@@ -612,18 +607,29 @@ def _add_condition(self, op, wire_list, col):
...
             # Add the open and closed buttons to indicate the condition value
-            for i in range(creg_size):
+            for i in range(creg_size - 1):
                 control = "\\control" if if_value[i] == "1" else "\\controlo"
                 self._latex[cwire + i][col] = f"{control} \\cw \\cwx[-" + str(gap) + "]"
                 gap = 1
+            # Add (hex condition value) below the last cwire
+            control = "\\control" if if_value[creg_size - 1] == "1" else "\\controlo"
+            self._latex[creg_size + cwire - 1][col] = (
+                f"{control}" + " \\cw^(%s){^{%s}} \\cwx[-%s]"
```
Comment (TharrmashasthaPV) [inline qiskit/visualization/latex.py:627]: The fix looks great! I was just curious to know if we could make a small aesthetic fix. In line 627, when we print the hex, it actually gets printed as a string inside a math environment. So it looks like 
image
If we could just change this line to
```
f"{control}" + " \\cw^(%s){^{\\mathtt{%s}}} \\cwx[-%s]"
```
then we can get something like
image
which I personally think looks more like a hexadecimal. Any thoughts on this?

Comment (enavarro51) [reply qiskit/visualization/latex.py:627]: Agreed. That x was a bit too fancy, and the new form is more compact as well. Will make the change.

**Review by TharrmashasthaPV — APPROVED** [review]
Perfect. LGTM.

**Review by 1ucian0 — APPROVED** [review]
Thanks!

**Pull request merged**